udmis/src/main/java/com/google/bos/udmi/monitoring/LogTail.java

```diff
@@ -0,0 +1,253 @@
+    } else {
+      String function_name = log.getResource().getLabels().getOrDefault("function_name", null);
+      if (function_name.equals("udmi_target")) {
+      } else if (function_name.equals("udmi_state")) {
+      } else if (function_name.equals("udmi_config")) {
+      }
+    }
+  }
+
+  /*
+   * Top level processor of incoming GCP log entries of ERROR type.
+   */
+  private void processLogEntryError(LogEntry log) {
+    // Be careful in here. Unwrapping this is a careful sequence.
+    Payload.ProtoPayload payload = log.getPayload();
+    if (!payload.getData().getTypeUrl().equals("type.googleapis.com/google.cloud.audit.AuditLog")) {
+      System.err.println("Not AuditLog");
+      return;
+    }
+    try {
+      AuditLog auditLog = AuditLog.parseFrom(payload.getData().getValue().toByteArray());
+      LogTailEntry entry = new LogTailEntry();
+      entry.timestamp = log.getInstantTimestamp();
+      entry.methodName = auditLog.getMethodName();
+      entry.serviceName = auditLog.getServiceName();
+      entry.resourceName = auditLog.getResourceName();
+      entry.statusCode = auditLog.getStatus().getCode();
+      entry.statusMessage = auditLog.getStatus().getMessage();
+      entry.severity = log.getSeverity();
+      logsTimeSeries.add(entry);
+      logsTimeSeries.maybeEmitMetrics();
+    } catch (InvalidProtocolBufferException e) {
+      System.err.println(e.toString());
+    } catch (IOException e) {
+      System.err.println(e.toString());
+    }
+  }
+
+  private void processLogStream(LogEntryServerStream stream) {
+    for (LogEntry log : stream) {
+      processLogEntry(log);
+    }
+  }
+
+  public void tailLogs() {
+    LogEntryServerStream stream = getCloudLogStream(LOG_FILTER);
+    processLogStream(stream);
+  }
+
+  /*
+   * Fields from GCP Logging stored in a neutral class.
+   */
+  protected class LogTailEntry {
+
+    Instant timestamp;
+    String methodName;
+    String serviceName;
+    String resourceName;
+    int statusCode;
+    String statusMessage;
+    Severity severity;
+    int value = 1;
+  }
+
+  /*
+   * Linked list of LogTailEntry.
+   */
+  protected class LogTailEntryList extends LinkedList<LogTailEntry> {
+
+  }
+
+  /*
+   * All the LogTail items currently processing.
+   * Hold the log entries in buckets for upstream processing in chunks.
+   */
+  protected class LogTimeSeries extends TreeMap<Long, LogTailEntryList> {
+
+    int numberOfLogs = 0;
+    int LOG_ENTRIES_BUCKET_SECONDS = 10;
+    int LOG_ENTRIES_PER_SUBMIT = 100;
+
+    private void emitJsonTimeSeries(PrintStream output) throws JsonProcessingException {
+      ObjectMapper objectMapper = new ObjectMapper();
+      for (long k : keySet()) {
+        for (LogTailEntry log : this.get(k)) {
+          Monitoring monitoring = new Monitoring();
+          monitoring.metric = new MonitoringMetric();
+          loadMetricFields(monitoring.metric, log);
+          output.println(objectMapper.writeValueAsString(monitoring));
```

Comment:
JsonUtils.stringify() -- sets some standard properties on the output mapper.

Reply:
Switched to using JsonUtil.stringify